contracts/misc/LensPeriphery.sol

```diff
@@ -27,15 +27,70 @@ contract LensPeriphery {
         keccak256(
             'ToggleFollowWithSig(uint256[] profileIds,bool[] enables,uint256 nonce,uint256 deadline)'
         );
+    bytes32 internal constant SET_PROFILE_METADATA_WITH_SIG_TYPEHASH =
+        keccak256(
+            'SetProfileMetadataURIWithSig(uint256 profileId,string metadata,uint256 nonce,uint256 deadline)'
+        );
 
-    ILensHub immutable HUB;
+    ILensHub public immutable HUB;
 
     mapping(address => uint256) public sigNonces;
 
+    mapping(address => mapping(uint256 => string)) internal _metadataByProfileByOwner;
+
     constructor(ILensHub hub) {
         HUB = hub;
     }
 
+    /**
+     * @notice Sets profile metadata for a profile owner as a dispatcher.
+     *
+     * @param profileId The profile ID to set the metadata for.
+     * @param metadata The metadata string to set for the profile and owner.
+     */
+    function dispatcherSetProfileMetadataURI(uint256 profileId, string calldata metadata) external {
+        address owner = IERC721Time(address(HUB)).ownerOf(profileId);
+        if (msg.sender != HUB.getDispatcher(profileId)) revert Errors.NotDispatcher();
```

Comment:
I was wondering if it's better to swap this two lines.

```suggestion
        if (msg.sender != HUB.getDispatcher(profileId)) revert Errors.NotDispatcher();
        address owner = IERC721Time(address(HUB)).ownerOf(profileId);
```

So if no dispatcher set, it does not even query the owner. 

Maybe this is put in this way so if the token does not exist, it will fail with the ownerOf message of nonexistent token instead of the NotDispatcher. But well, I point this just in case you think worth the change. Otherwise just merge, code looks good!

Reply:
I think in this specific case, it makes more sense for the end user to receive the revert message of the profile not existing. This is a valid optimization insofar as the call reverts because the user is not the dispatcher, but I think odds of calling this with an incorrect profile ID are more likely, in which case the opposite is more optimal.